app/models/slot_availability.rb

```diff
@@ -0,0 +1,14 @@
+class SlotAvailability
+  include NonPersistedModel, Comparable
+  delegate :begin_at, :iso8601, to: :slot
+  attribute :slot, ConcreteSlot
+  attribute :unavailability_reasons, Array[String], default: []
+
+  def bookable?
+    unavailability_reasons.empty?
+  end
+
+  def <=>(other)
+    slot <=> other.slot
```

Comment:
Do we have `<=>` defined in `ConcreteSlot`? I added it to pvb2, just not sure if public has it.